DigitalLearningSolutions.Data.Tests/DataServices/RegistrationDataServiceTests.cs

```diff
@@ -195,6 +195,181 @@ public async Task
+
+            // Given
+            var delegateRegistrationModel =
+                RegistrationModelTestHelper.GetDefaultDelegateRegistrationModel(
+                    "forename",
+                    "surname",
+                    "test@gmail.com",
+                    active: true,
+                    approved: false
+                );
+            var currentTime = new DateTime(2022, 06, 27, 11, 03, 12);
+            const int userId = 281052;
+            const int existingDelegateId = 142559;
+
+            // When
+            var userBeforeUpdate = userDataService.GetUserAccountById(userId);
+            var delegateBeforeUpdate = userDataService.GetDelegateAccountById(existingDelegateId);
+            service.ReregisterDelegateAccountAndCentreDetailForExistingUser(
+                delegateRegistrationModel,
+                userId,
+                existingDelegateId,
+                currentTime
+            );
+
+            // Then
+            var userAfterUpdate = userDataService.GetUserAccountById(userId);
+            var delegateAfterUpdate = userDataService.GetDelegateAccountById(existingDelegateId);
+            using (new AssertionScope())
+            {
+                var oldDateRegistered = new DateTime(2014, 12, 24, 10, 44, 53, 257);
+                var oldCentreSpecificDetailsLastChecked = new DateTime(2022, 04, 27, 16, 31, 29, 897);
+                userBeforeUpdate.Should().NotBeNull();
+                delegateBeforeUpdate.Should().NotBeNull();
+                userAfterUpdate.Should().NotBeNull();
+                delegateAfterUpdate.Should().NotBeNull();
+                userBeforeUpdate!.Id.Should().Be(userId);
+                userBeforeUpdate.FirstName.Should().Be("xxxx");
+                userBeforeUpdate.LastName.Should().Be("xxxx");
+                userBeforeUpdate.PrimaryEmail.Should().Be("noe");
+                userBeforeUpdate.DetailsLastChecked.Should().BeNull();
+                userBeforeUpdate.JobGroupId.Should().Be(4);
+                userBeforeUpdate.PasswordHash.Should().BeEmpty();
+                userBeforeUpdate.ProfessionalRegistrationNumber.Should().BeNull();
+
+                userAfterUpdate!.Id.Should().Be(userBeforeUpdate.Id);
+                userAfterUpdate.FirstName.Should().Be(userBeforeUpdate.FirstName);
+                userAfterUpdate.LastName.Should().Be(userBeforeUpdate.LastName);
+                userAfterUpdate.PrimaryEmail.Should().Be(userBeforeUpdate.PrimaryEmail);
+                userAfterUpdate.DetailsLastChecked.Should().Be(userBeforeUpdate.DetailsLastChecked);
+                userAfterUpdate.JobGroupId.Should().Be(userBeforeUpdate.JobGroupId);
+                userAfterUpdate.PasswordHash.Should().Be(userBeforeUpdate.PasswordHash);
+                userAfterUpdate.ProfessionalRegistrationNumber.Should()
+                    .Be(userBeforeUpdate.ProfessionalRegistrationNumber);

```

Comment:
It looks like we're checking that all(?) the properties of `userAfterUpdate` are the same as those in `userBeforeUpdate`. Is there a simple one-line test we can do to test this instead? 

The above code made me wonder if there are some other properties that we aren't testing that might have changed.

It actually seems to me that we don't really care about the property values of `userBeforeUpdate` or `userAfterUpdate` at all, just that they are the same, and not null.